### PR TITLE
Set properties in Core.DSExecutable

### PR DESCRIPTION
**Pull request by zora-wang**
@junmendoza 

ps: 24 failures with the following files changes.

### CODE REVIEW

**Comment by junmendoza** [comment]
@zora-wang can you remove the commented code?

File: src/Engine/ProtoAssociative/CodeGen.cs [hunk 1]
```diff
@@ -4003,8 +4003,8 @@ public override int Emit(ProtoCore.AST.Node codeBlockNode, ProtoCore.Associative
                             this.core.ClassTable.AuditMultipleDefinition(this.core.BuildStatus, graphNode);
                         }
                         codeblock.Body = BuildSSA(codeblock.Body, context);
-                        core.DSExecutable.CachedSSANodes.Clear();
-                        core.DSExecutable.CachedSSANodes.AddRange(codeblock.Body);
+                        //core.DSExecutable.CachedSSANodes.Clear();
```
Comment (junmendoza) [inline src/Engine/ProtoAssociative/CodeGen.cs]: Remove the commented code

Comment (zora-wang) [inline src/Engine/ProtoAssociative/CodeGen.cs]: He @junmendoza , The comments have been removed ~~~

**Comment by junmendoza** [comment]
@zora-wang I have found the problem with this.
Resetting DSExecutable also resetted properties such as CallsiteCache that are used by the Runtime.
1. We should move RuntimeData property from Core to RuntimeCore 
2. Move the following properties from DSExecutable to RuntimeData: CallsiteCache, CallSiteToNodeMap, ASTToCallSiteMap

**Comment by junmendoza** [comment]
@zora-wang Thanks
FYI this PR is to address the previous reverted PR: https://github.com/DynamoDS/Dynamo/pull/4277

**Comment by junmendoza** [comment]
@zora-wang I will take care of the revit changes

**Pull request merged**